src/slot_provider/slot.rs

```diff
@@ -0,0 +1,66 @@
+use std::{fmt::Display, ops::Add};
+
+use chrono::{DateTime, Duration, Utc};
+use eyre::Result;
+use lazy_static::lazy_static;
+
+lazy_static! {
+    pub static ref BEACON_GENESIS: DateTime<Utc> = "2020-12-01T12:00:23Z".parse().unwrap();
+}
+
+/// A slot number on the beacon chain.
+/// Started at 0 at 2020-12-01T12:00:23Z (genesis). Slots follow unix timestamps meaning most slots
+/// are 12 seconds long, but some are 13 or 11 seconds long. We use u32 to store. Enough for 12
+/// seconds * 2^62 = ~1634 years.
+#[derive(Clone, Copy, Debug)]
+pub struct Slot(pub u32);
+
+impl Slot {
+    // Note: this is true virtually all of the time but because of leap seconds not always.
+    pub const SLOT_PERIOD_SECONDS: u64 = 12;
+
+    /// Create a slot from a timestamp.
+    /// Rounds down to the nearest slot. i.e. what slot was the timestamp in?
+    pub fn from_date_time_round_down(date_time: DateTime<Utc>) -> Result<Slot> {
+        if date_time < *BEACON_GENESIS {
+            eyre::bail!(
+                "cannot convert DateTime ({}) before beacon genesis ({}) to Slot",
+                date_time,
+                *BEACON_GENESIS
+            )
+        }
+
+        let slot = Slot(
+            ((date_time - *BEACON_GENESIS).num_seconds() / Self::SLOT_PERIOD_SECONDS as i64) as u32,
+        );
+        Ok(slot)
+    }
+
+    pub fn to_date_time(self) -> DateTime<Utc> {
+        *BEACON_GENESIS + Duration::seconds(self.0 as i64 * Self::SLOT_PERIOD_SECONDS as i64)
+    }
+
+    pub fn now() -> Self {
+        Self::from_date_time_round_down(Utc::now()).expect("Expect now to be after beacon genesis")
+    }
+}
+
+impl From<Slot> for DateTime<Utc> {
+    fn from(slot: Slot) -> Self {
+        *BEACON_GENESIS + Duration::seconds(slot.0 as i64 * Slot::SLOT_PERIOD_SECONDS as i64)
+    }
+}
+
+impl Add<u32> for Slot {
```

Comment:
Perhaps instead of this, which I only added to be able to do:
```rust
let current_slot = Slot::now();
let next_slot = current_slot + 1;
```

We should just have a `.next()` method. After all, when would you ever add any number but 1?

Reply:
That'd make sense.